erpnext/accounts/doctype/pos_invoice_merge_log/pos_invoice_merge_log.py

```diff
@@ -55,42 +55,61 @@ def validate_pos_invoice_status(self):
 	def on_submit(self):
 		pos_invoice_docs = [frappe.get_doc("POS Invoice", d.pos_invoice) for d in self.pos_invoices]
 
-		returns = [d for d in pos_invoice_docs if d.get('is_return') == 1]
 		sales = [d for d in pos_invoice_docs if d.get('is_return') == 0]
+		returns = [d for d in pos_invoice_docs if d.get('is_return') == 1]
+		advances, cn_amount_mapping = get_pos_advance_credit_notes(sales, returns)
 
-		sales_invoice, credit_note = "", ""
+		sales_invoice, credit_note, advance_credit_note, advance_credit_note_doc = "", "", "", ""
 		if returns:
 			credit_note = self.process_merging_into_credit_note(returns)
 
+		if advances:
+			advance_credit_note = self.process_merging_into_credit_note(advances, is_advance=True)
+			advance_credit_note_doc = frappe.get_doc("Sales Invoice", advance_credit_note)
```

Comment:
This `get_doc` can be eliminated, if you return `advance_credit_note` doc from `process_merging_into_credit_note`